utils/yoke/src/trait_hack.rs

```diff
@@ -26,53 +26,100 @@
 //!
 //! # Examples
 //!
-//! Code that does not compile:
+//! Code that does not compile ([playground](https://play.rust-lang.org/?version=beta&mode=debug&edition=2018&gist=ebbda5b15a398d648bdff9e439b27dc0)):
 //!
-//! ***[#1061](https://github.com/unicode-org/icu4x/issues/1061): The following example
-//! compiles starting in Rust 1.56.
+//! ```compile_fail
+//! trait MiniYokeable<'a> {
```

Comment:
Let's not check in MiniYoke stuff: can we get the version of this that uses actual Yoke?

Reply:
I was afraid you'd ask that... I copied my minified example into here because it serves the purpose of the `compile_fail`.  But I'll try to un-minify it...

Reply:
Yeah I'd like this to show off the kind of code that will actually break in ICU4X :)

Reply:
ok so note that this is in the `yoke` crate, not the `icu_provider` crate.  So I updated the example to use the real Yoke/Yokeable but MiniDataProvider.  Is that ok or do you want me to add the dev dependency, or move this to the other crate?